q2_fragment_insertion/_format.py

```diff
@@ -6,16 +6,86 @@
 # The full license is in the file LICENSE, distributed with this software.
 # ----------------------------------------------------------------------------
 
+import re
+
+import ijson
+import skbio
+
 import qiime2.plugin.model as model
+from qiime2.plugin import ValidationError
+
+from q2_types.feature_data import AlignedDNAFASTAFormat
+from q2_types.tree import NewickFormat
 
 
 class PlacementsFormat(model.TextFileFormat):
-    def sniff(self):
-        line = open(str(self)).readline()
+    fields = {'tree', 'placements', 'metadata', 'version', 'fields'}
+
+    def _validate_(self, level):
+        keys_found = set()
 
-        # it's json... but would be very expensive to parse
-        return line[0] == '{'
+        # Can't self.open(mode='rb'), so we defer to the backing pathlib object
+        with self.path.open(mode='rb') as fh:
+            root_element = None
+            for prefix, event, value in ijson.parse(fh):
+                if root_element is None:
+                    if event != 'start_map':
+                        raise ValidationError('Root element of file must be a '
+                                              'JSON object')
+                    else:
+                        root_element = True
+
+                # Skip parsing attributes that could be prohibitively large
+                if prefix.startswith('placements') \
+                        or prefix.startswith('tree'):
+                    continue
+
+                # Restricted to only checking root-level keys
+                if event == 'map_key' and prefix == '':
+                    keys_found.add(value)
+
+        if keys_found != self.fields:
+            raise ValidationError('Expected the following fields: %s, found '
+                                  '%s.' % (sorted(self.fields),
+                                           sorted(keys_found)))
 
 
 PlacementsDirFmt = model.SingleFileDirectoryFormat(
     'PlacementsDirFmt', 'placements.json', PlacementsFormat)
+
+
+class RAxMLinfoFormat(model.TextFileFormat):
+    def _validate_(self, level):
+        sigs = ['This is RAxML version', 'Base frequencies',
+                'Final GAMMA likelihood']
+
+        info = self.path.read_text()
+
+        for sig in sigs:
+            new_sig = sig.replace(r' ', r'\W+')
+            if not re.search(new_sig, info):
+                raise ValidationError('Missing structured content: "%s".'
+                                      % sig)
+
+
+class SeppReferenceFormat(model.DirectoryFormat):
+    alignment = model.File(r'aligned-dna-sequences.fasta',
+                           format=AlignedDNAFASTAFormat)
+    phylogeny = model.File(r'tree.nwk', format=NewickFormat)
+    raxml_info = model.File(r'raxml-info.txt', format=RAxMLinfoFormat)
+
```

Comment:
didn't we want to add default values for e.g. alignment_subset_size to the reference qza?

Reply:
Maybe? I was waiting to hear back from you on that:

> Based on the comments in the code, my understanding is that the default values are specific to GG. If that is the case, then doesn't it make sense to require users specify that value themselves, that way they use the values relevant to the DB in question? I am curious on your thoughts regarding that - thanks.

My initial thought was "yes", but then I noticed in #63 you didn't make any notes about these values changing for Silva (or if you did, I missed them). Can you comment on this?

Reply:
for now, I don't have those values for Silva or other references. I hope to find a student volunteer soon to benchmark Silva. Thus, I agree with your assessment and vote against moving those parameters to the reference set

Reply:
that said: is there a way to overwrite regular parameters for a qiime2 function, i.e. can we optionally provide better suiting default per reference in the future? Thus, the normal user would not need to worry about both parameters